crates/symbolicator/src/cache.rs

```diff
@@ -26,6 +26,37 @@ use crate::types::Scope;
 /// yet.
 pub const MALFORMED_MARKER: &[u8] = b"malformed";
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum MalformedCause {
+    BadObject,
+    DownloadError,
```

Comment:
How bad would it be to go straight to `DownloadError(String)` (same for `BadObject`)?  That way we can even add a message.

Reply:
So I'm a little hesitant to commit to storing a full string here for two reasons:
1. It would make reading the malformed file a little bit more difficult, and easier to game/fool since the length of the file would be variable at this point.
2. The dream would be to figure out a way to restore the underlying download or object error based off of this string, which `DownloadError(String)` doesn't really solve ): Key places where it would be useful to restore the original error when we have access to only the string read from the file would be in `SymCacheFile::parse()` and `CfiCacheModule::extend()`, both of which are changed in this diff. 

I figured a good compromise was to store the general category of error without preserving specifics via this enum. The malformed file's contents continue to be restricted to a very small, predictable set of values at the cost of losing some detail. Something to note is that this does loosely relate to and/or address the comment above as well. 

I'm open to being convinced that using `BadObject(String)`, `DownloadError(String)`, `Other(String)` might be a better idea. Alternative approaches where we could store a string that allows for a more meaningful conversion from `CacheStatus` to `ObjectFileStatus` would be very welcome, since the latter does have a variant that wraps around `DownloadError`s and whatnot as well.

Reply:
see https://github.com/getsentry/symbolicator/pull/502#discussion_r677972226